src/sentry/migrations/0507_delete_pending_deletion_rules.py

```diff
@@ -10,9 +10,10 @@ def delete_rules(apps, schema_editor):
     from sentry.constants import ObjectStatus
 
     Rule = apps.get_model("sentry", "Rule")
+    RegionScheduledDeletion = apps.get_model("sentry", "RegionScheduledDeletion")
     for rule in RangeQuerySetWrapperWithProgressBar(Rule.objects.all()):
         if rule.status in (ObjectStatus.PENDING_DELETION, ObjectStatus.DISABLED):
-            rule.delete()
+            RegionScheduledDeletion.schedule(rule, days=0)
```

Comment:
That probably won't work, since you don't have a real model here and so none of our custom functions exist. You can copy the code from `schedule` into a separate function in this file and just call it directly with the rule